### PR TITLE
Fix various typos in doc

### PR DESCRIPTION
**Pull request by yannickmoy**

### CODE REVIEW

File: docs/DafnyRef/Statements.md [hunk 1]
```diff
@@ -464,6 +464,7 @@ if tmp.IsFailure() {
   rr := tmp.PropagateFailure();
   return;
 }
+j := tmp.Extract();
```
Comment (davidcok) [inline docs/DafnyRef/Statements.md:467]: This is incorrect. A Status value has no Extract (whereas an Outcome does)

Comment (davidcok) [reply docs/DafnyRef/Statements.md:467]: But line 454 should be
  k :- r, 7;

Comment (yannickmoy) [reply docs/DafnyRef/Statements.md]: ah right. updated the patch

Comment (yannickmoy) [reply docs/DafnyRef/Statements.md]: ok, so j is useless, see my latest patch

**Pull request merged**